### PR TITLE
feat: Add Enable and Disable endpoints

### PR DESCRIPTION
**Pull request by jachym-tousek-keboola**
Jira: https://keboola.atlassian.net/browse/PSGO-675

**Changes:**
- Added endpoints to enable/disable source/sink.

-----------

### CODE REVIEW

File: internal/pkg/service/stream/api/service/sink.go [hunk 1]
```diff
@@ -306,6 +306,22 @@ func (s *service) SinkStatisticsClear(ctx context.Context, d dependencies.SinkRe
 	return d.StatisticsRepository().ResetSinkStats(d.SinkKey()).Do(ctx).Err()
 }
 
+func (s *service) DisableSink(ctx context.Context, d dependencies.SinkRequestScope, payload *api.DisableSinkPayload) (err error) {
+	if err := s.sinkMustExists(ctx, d.SinkKey()); err != nil {
+		return err
+	}
+
+	return s.definition.Sink().Disable(d.SinkKey(), d.Clock().Now(), d.RequestUser(), "TODO").Do(ctx).Err()
```
[the remark this answers is not in the document]
Comment (michaljurecko) [reply internal/pkg/service/stream/api/service/sink.go:314]: @jachym-tousek-keboola You can put some constant there: `manually`, `API`, ...

It is prepared for automatic disabling of non-functional streams, e.g. when the Storage API token is refreshed, or for some other reason it does not work for a long time.

File: api/stream/design.go [hunk 2]
```diff
@@ -302,6 +302,32 @@ var _ = Service("stream", func() {
 		})
 	})
 
+	Method("DisableSource", func() {
+		Meta("openapi:summary", "Disable source")
+		Description("Disables the source.")
+		Result(Task)
+		Payload(GetSourceRequest)
+		HTTP(func() {
+			POST("/branches/{branchId}/sources/{sourceId}/disable")
```
Comment (Matovidlo) [inline api/stream/design.go:311]: Shouldn't this be a `PUT` request? We modify existing entity.

File: test/stream/api/sink/create-014-disable-enable/expected-etcd-kvs.txt [hunk 3]
```diff
@@ -0,0 +1,231 @@
...
+  "branchId": %%TEST_DEFAULT_BRANCH_ID%%,
+  "sourceId": "my-source",
+  "sinkId": "my-sink",
+  "token": {
+    %A,
+    "bucketPermissions": {
+      "in.c-my-bucket": "write"
+    }
+  }
+}
```
Comment (Matovidlo) [inline test/stream/api/sink/create-014-disable-enable/expected-etcd-kvs.txt:122]: This could be `%A` as we do not need to test secrets in this test.

**Review by hosekpeter — APPROVED** [review]

**Review by Matovidlo — APPROVED** [review]
LGTM

**Pull request merged**